docs/pages/kubernetes-access/getting-started/local.mdx

```diff
@@ -0,0 +1,349 @@
+mobile device. You will use this to authenticate to your Teleport cluster.
+
+## Step 1/4 Deploy resources
+
+### Start minikube
+
+Start minikube with the Docker driver:
+
+```code
+$ minikube start --driver=docker
+```
+
+Add the Teleport Helm repository. Our Helm charts make it easier to deploy
+Teleport on your local Kubernetes cluster:
+
+```code
+$ helm repo add teleport https://charts.releases.teleport.dev
+```
+
+### Install the Teleport Auth Service and Proxy Service
+
+You will deploy the Auth Service and Proxy Service by installing the
+`teleport-cluster` Helm chart. To do so, run the following commands:
+
+```code
+# This is the DNS name Kubernetes will assign to the Proxy Service
+$ CLUSTER_NAME="teleport-cluster.teleport-cluster.svc.cluster.local"
+$ helm install teleport-cluster teleport/teleport-cluster \
+--create-namespace \
+--namespace=teleport-cluster \
+--set clusterName=${CLUSTER_NAME?}
+$ kubectl config set-context --current --namespace teleport-cluster
+```
+
+Any `kubectl` commands you run will now use the `teleport-cluster` namespace. 
+
+Verify that Teleport is running.
+
+```code
+$ kubectl get pods
+NAME                               READY   STATUS    RESTARTS   AGE
+teleport-cluster-b9dc6c68b-d75fb   1/1     Running   0          46s
+```
+
+### Expose the Proxy Service to your local machine
+
+The `teleport-cluster` service is of type `LoadBalancer`, meaning that your
+platform needs to supply a load balancer to route traffic to the service.
+
+The `minikube tunnel` command provides a load balancer and establishes a reverse
+SSH tunnel to ports on your local machine.
+
+Open a new terminal and run the following command to create a reverse tunnel. In
+this case, the tunnel forwards port 443, so you will need to enter your
+password.
+
+```code
+$ minikube tunnel
+✅  Tunnel successfully started
+
+📌  NOTE: Please do not close this terminal as this process must stay alive for the tunnel to be accessible ...
+
+❗  The service/ingress teleport-cluster requires privileged ports to be exposed: [443]
+🔑  sudo permission will be asked for it.
+🏃  Starting tunnel for service teleport-cluster.
+Password:
+```
+
+The `teleport-cluster` service should now have an external IP:
+
+```code
+$ kubectl get services
+NAME               TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                                                                     AGE
+teleport-cluster   LoadBalancer   10.107.218.212   127.0.0.1     443:32143/TCP,3023:30618/TCP,3026:32750/TCP,3024:32406/TCP,3036:30687/TCP   6m18s
+```
+
+The Proxy Service enables you to manage your cluster via an HTTP API. Assign the
+external IP of the load balancer to an environment variable and verify that the
+HTTP API is working:
+
+```code
+$ EXTERNAL_IP=$(kubectl get service teleport-cluster -o jsonpath='{ .status.loadBalancer.ingress[0].ip }')
+$ curl --insecure https://${EXTERNAL_IP?}:443/webapi/ping
```

Comment:
Did you or whoever wrote these instructions test this on an M1 Mac?  Most likely I did something wrong, but I get the following output:

% kubectl get services  
NAME               TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)                                                                     AGE
teleport-cluster   LoadBalancer   10.98.32.218   127.0.0.1     443:30951/TCP,3023:31431/TCP,3026:30121/TCP,3024:32661/TCP,3036:31735/TCP   6m1s
% curl --insecure https://${EXTERNAL_IP?}:443/webapi/ping
curl: (7) Failed to connect to 127.0.0.1 port 443 after 6 ms: Connection refused

I think this may be related to the minikube tunnel problem mentioned at https://github.com/kubernetes/minikube/issues/13510

Reply:
The `curl` command worked for me after following the instructions on my M1.

Can you confirm whether the SSH reverse tunnel that minikube creates to the host in `minikube tunnel` is running?

```
$ sudo lsof -i -P | grep "localhost:443"
ssh       52154            root    5u  IPv6 0xef95ab8bc066e1a1      0t0    TCP localhost:443 (LISTEN)
ssh       52154            root    6u  IPv4 0xef95ab8bc6403d39      0t0    TCP localhost:443 (LISTEN)
```

Reply:
The SSH reverse tunnel is not running.  `sudo lsof -i -P | grep "localhost:443"` returns no output.  

`minikube tunnel --alsologtostderr` just shows:
```
I0322 18:18:05.661218   53280 out.go:176] 🏃  Starting tunnel for service teleport-cluster.
🏃  Starting tunnel for service teleport-cluster.
```

And then it just sits there.

I can do this though:
```
%kubectl port-forward service/teleport-cluster 443:30951
error: Service teleport-cluster does not have a service port 30951

%kubectl port-forward service/teleport-cluster 30951:443
Forwarding from 127.0.0.1:30951 -> 3080
Forwarding from [::1]:30951 -> 3080
Handling connection for 30951

%curl --insecure https://127.0.0.1:30951/webapi/ping
{"auth":{"type":"local","second_factor":"on","preferred_local_mfa":"webauthn","webauthn":{"rp_id":"teleport-cluster.teleport-cluster.svc.cluster.local"},"has_motd":false},"proxy":{"kube":{"enabled":true,"listen_addr":"0.0.0.0:3026"},"ssh":{"listen_addr":"[::]:3023","tunnel_listen_addr":"0.0.0.0:3024","public_addr":"teleport-cluster.teleport-cluster.svc.cluster.local:443"},"db":{"mysql_listen_addr":"0.0.0.0:3036"},"tls_routing_enabled":false},"server_version":"9.0.1","min_client_version":"8.0.0"}%
```

Reply:
What do your logs say when using `--alsologtostderr`? I am able to run `minikube tunnel` in one terminal and the `curl` command against the `/webapi/ping` path in another. I get these logs in `~/Desktop/stderr.log` if I run `minikube tunnel --alsologtostderr 2>~/Desktop/stderr.log`:

```
I0324 15:57:58.362694   29491 out.go:297] Setting OutFile to fd 1 ...
I0324 15:57:58.362881   29491 out.go:349] isatty.IsTerminal(1) = true
I0324 15:57:58.362889   29491 out.go:310] Setting ErrFile to fd 2...
I0324 15:57:58.362896   29491 out.go:349] isatty.IsTerminal(2) = false
I0324 15:57:58.362982   29491 root.go:315] Updating PATH: /Users/paulgottschling/.minikube/bin
I0324 15:57:58.363535   29491 mustload.go:65] Loading cluster: minikube
I0324 15:57:58.364005   29491 config.go:176] Loaded profile config "minikube": Driver=docker, ContainerRuntime=docker, KubernetesVersion=v1.23.1
I0324 15:57:58.364541   29491 cli_runner.go:133] Run: docker container inspect minikube --format={{.State.Status}}
I0324 15:57:58.479609   29491 host.go:66] Checking if "minikube" exists ...
I0324 15:57:58.479896   29491 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "8443/tcp") 0).HostPort}}'" minikube
I0324 15:57:58.508014   29491 api_server.go:165] Checking apiserver status ...
I0324 15:57:58.508117   29491 ssh_runner.go:195] Run: sudo pgrep -xnf kube-apiserver.*minikube.*
I0324 15:57:58.508180   29491 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0324 15:57:58.536689   29491 sshutil.go:53] new ssh client: &{IP:127.0.0.1 Port:53644 SSHKeyPath:/Users/paulgottschling/.minikube/machines/minikube/id_rsa Username:docker}
I0324 15:57:58.626145   29491 ssh_runner.go:195] Run: sudo egrep ^[0-9]+:freezer: /proc/1653/cgroup
I0324 15:57:58.633069   29491 api_server.go:181] apiserver freezer: "7:freezer:/docker/f685093ac9941de75c1fdae77096478791f5f2239cf610c809a2719a1e69965b/kubepods/burstable/pod96be69ce9ff7dc0acff6fda2873a009a/82616683d4e097e9fdd680fdac6a630b36cbdb7c8d595dfbc5dbadf97fa753f2"
I0324 15:57:58.633153   29491 ssh_runner.go:195] Run: sudo cat /sys/fs/cgroup/freezer/docker/f685093ac9941de75c1fdae77096478791f5f2239cf610c809a2719a1e69965b/kubepods/burstable/pod96be69ce9ff7dc0acff6fda2873a009a/82616683d4e097e9fdd680fdac6a630b36cbdb7c8d595dfbc5dbadf97fa753f2/freezer.state
I0324 15:57:58.639515   29491 api_server.go:203] freezer state: "THAWED"
I0324 15:57:58.639552   29491 api_server.go:240] Checking apiserver healthz at https://127.0.0.1:53643/healthz ...
I0324 15:57:58.644345   29491 api_server.go:266] https://127.0.0.1:53643/healthz returned 200:
ok
I0324 15:57:58.644374   29491 tunnel.go:59] Checking for tunnels to cleanup...
I0324 15:57:58.647875   29491 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0324 15:57:58.697763   29491 out.go:176] * Tunnel successfully started
I0324 15:57:58.716563   29491 out.go:176] 
I0324 15:57:58.737033   29491 out.go:176] * NOTE: Please do not close this terminal as this process must stay alive for the tunnel to be accessible ...
I0324 15:57:58.756971   29491 out.go:176] 
I0324 15:57:58.787027   29491 out.go:176] ! The service/ingress teleport-cluster requires privileged ports to be exposed: [443]
I0324 15:57:58.806183   29491 out.go:176] * sudo permission will be asked for it.
I0324 15:57:58.810833   29491 loadbalancer_patcher.go:122] Patched teleport-cluster with IP 127.0.0.1
I0324 15:57:58.841431   29491 out.go:176] * Starting tunnel for service teleport-cluster.

```

Reply:
Here's what I get:
```
I0324 16:56:44.173533    9798 out.go:297] Setting OutFile to fd 1 ...
I0324 16:56:44.173924    9798 out.go:349] isatty.IsTerminal(1) = true
I0324 16:56:44.173932    9798 out.go:310] Setting ErrFile to fd 2...
I0324 16:56:44.173939    9798 out.go:349] isatty.IsTerminal(2) = false
I0324 16:56:44.173993    9798 root.go:315] Updating PATH: /Users/ulysses/.minikube/bin
W0324 16:56:44.174068    9798 root.go:293] Error reading config file at /Users/ulysses/.minikube/config/config.json: open /Users/ulysses/.minikube/config/config.json: no such file or directory
I0324 16:56:44.174299    9798 mustload.go:65] Loading cluster: minikube
I0324 16:56:44.174629    9798 config.go:176] Loaded profile config "minikube": Driver=docker, ContainerRuntime=docker, KubernetesVersion=v1.23.3
I0324 16:56:44.175008    9798 cli_runner.go:133] Run: docker container inspect minikube --format={{.State.Status}}
I0324 16:56:44.383817    9798 host.go:66] Checking if "minikube" exists ...
I0324 16:56:44.384186    9798 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "8443/tcp") 0).HostPort}}'" minikube
I0324 16:56:44.502585    9798 api_server.go:165] Checking apiserver status ...
I0324 16:56:44.502714    9798 ssh_runner.go:195] Run: sudo pgrep -xnf kube-apiserver.*minikube.*
I0324 16:56:44.502786    9798 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0324 16:56:44.616599    9798 sshutil.go:53] new ssh client: &{IP:127.0.0.1 Port:49948 SSHKeyPath:/Users/ulysses/.minikube/machines/minikube/id_rsa Username:docker}
I0324 16:56:44.713387    9798 ssh_runner.go:195] Run: sudo egrep ^[0-9]+:freezer: /proc/1497/cgroup
W0324 16:56:44.721326    9798 api_server.go:176] unable to find freezer cgroup: sudo egrep ^[0-9]+:freezer: /proc/1497/cgroup: Process exited with status 1
stdout:

stderr:
I0324 16:56:44.721360    9798 api_server.go:240] Checking apiserver healthz at https://127.0.0.1:49952/healthz ...
I0324 16:56:44.727132    9798 api_server.go:266] https://127.0.0.1:49952/healthz returned 200:
ok
I0324 16:56:44.727163    9798 tunnel.go:59] Checking for tunnels to cleanup...
I0324 16:56:44.728843    9798 cli_runner.go:133] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0324 16:56:44.861454    9798 out.go:176] * Tunnel successfully started
I0324 16:56:44.881185    9798 out.go:176] 
I0324 16:56:44.900140    9798 out.go:176] * NOTE: Please do not close this terminal as this process must stay alive for the tunnel to be accessible ...
I0324 16:56:44.920333    9798 out.go:176] 
I0324 16:56:44.949734    9798 out.go:176] ! The service/ingress teleport-cluster requires privileged ports to be exposed: [443]
I0324 16:56:44.969045    9798 out.go:176] * sudo permission will be asked for it.
I0324 16:56:45.004080    9798 out.go:176] * Starting tunnel for service teleport-cluster.
I0324 16:56:44.976407    9798 loadbalancer_patcher.go:122] Patched teleport-cluster with IP 127.0.0.1
```

Main differences compared to yours are the missing minikube config file and also the `unable to find freezer cgroup`.  Also, the ordering of patching the cluster with IP and starting the tunnel is different.  Not sure these items would cause what I'm seeing where I get `curl: (7) Failed to connect to 127.0.0.1 port 443 after 4 ms: Connection refused` though.  I guess we could see what you have in your minikube config file?

Reply:
I dug into this more, and it looks like `minikube tunnel` opens an SSH reverse tunnel [via running an `ssh` command](https://github.com/kubernetes/minikube/blob/3493e13d100582359cf94aae5d5958718e826b57/pkg/minikube/tunnel/kic/ssh_conn.go#L96-L97) from your host machine. 

Maybe try showing the output of the SSH command? E.g., clone `kubernetes/minikube`, then add these two lines to attach the command's stderr to the `minikube` process's stderr and enable verbose logging (all in [this function](https://github.com/kubernetes/minikube/blob/3493e13d100582359cf94aae5d5958718e826b57/pkg/minikube/tunnel/kic/ssh_conn.go#L41)):

```diff
--- a/pkg/minikube/tunnel/kic/ssh_conn.go
+++ b/pkg/minikube/tunnel/kic/ssh_conn.go
@@ -42,6 +42,7 @@ func createSSHConn(name, sshPort, sshKey string, resourcePorts []int32, resource
        // extract sshArgs
        sshArgs := []string{
                // TODO: document the options here
+               "-vvv", // enable verbose logging
                "-o", "UserKnownHostsFile=/dev/null",
                "-o", "StrictHostKeyChecking=no",
                "-N",
@@ -88,6 +89,7 @@ func createSSHConn(name, sshPort, sshKey string, resourcePorts []int32, resource
        }
 
        cmd := exec.Command(command, sshArgs...)
+       cmd.Stderr = os.Stderr // Attach the ssh subprocess's stderr to the minikube process's stderr
 
        return &sshConn{
                name:       name,
```

Then run `make minikube-darwin-arm64` in the minikube clone and run `minikube tunnel` using the path to the new binary. If there's an issue with the SSH connection, hopefully it will show up there. For me, my `ssh` debug logs ended in:

```
debug1: channel 10: free: direct-tcpip: listening port 443 for 10.102.226.241 port 443, connect from 127.0.0.1 port 54828 to 127.0.0.1 port 443, nchannels 11
debug3: channel 10: status: The following connections are open:
  #10 direct-tcpip: listening port 443 for 10.102.226.241 port 443, connect from 127.0.0.1 port 54828 to 127.0.0.1 port 443 (t4 r0 i3/0 o3/0 e[closed]/0 fd 15/15/-1 sock 15 cc -1)
```

 What's weird, though, is that if there were an error starting the SSH tunnel, you'd see [this log](https://github.com/kubernetes/minikube/blob/3493e13d100582359cf94aae5d5958718e826b57/pkg/minikube/tunnel/kic/ssh_tunnel.go#L133-L134), which doesn't come up. 
 
 I'm not sure the minikube config is an issue, since I'm using the defaults.
 
 We can meet about this next week if that works for you! I can also see if it makes sense to talk about `kubectl port-forward` as an alternative.

Reply:
Good idea to add SSH debugging.  I did so and built minikube, and get this at the end:

```
Received disconnect from 127.0.0.1 port 56814:2: Too many authentication failures
Disconnected from 127.0.0.1 port 56814
```

So that appears to be the problem.  Sure, we can meet next week and see if there's anything further that can be done.  Since everyone else is getting this to work, this looks like something unique to my (mis)configuration.

Reply:
Do the logs show what authentication methods SSH is trying? I can add an `Admonition` or `Notice` about `minikube tunnel` and a user's local SSH configuration. Since `minikube` just shells out `ssh`, it's likely that other readers will run into the same issue!

Reply:
I took a closer look at the log.  It listed 8 keys to try, but when it got to the 5th one, it gave the `Too many authentication failures` error.  I resolved this by clearing the list of identities from the agent with `ssh-add -D` and then my `.minikube/machines/minikube/id_rsa` was accepted.  The tunnel + the curl worked fine as documented.  Hooray!  

I support you could add a notice about the error above and reducing the number of identities in your SSH agent to less than 5.  Though people wouldn't actually see the error, just a hanging tunnel ...

Reply:
Thanks for taking the time to help troubleshoot! I've added a `Details` box with troubleshooting steps